csharp/src/SeedLang/Ast/Executor.cs

```diff
@@ -73,6 +73,57 @@ internal sealed class Executor : AstWalker {
       }
     }
 
+    protected override void Visit(CompareExpression compare) {
+      Visit(compare.First);
+      IValue first = _expressionResult;
+      var exprs = new IValue[compare.Exprs.Length];
+      bool currentResult = true;
+      for (int i = 0; i < compare.Ops.Length; ++i) {
+        Visit(compare.Exprs[i]);
+        exprs[i] = _expressionResult;
+        IValue left = i > 0 ? exprs[i - 1] : first;
+        switch (compare.Ops[i]) {
+          case CompareOperator.Less:
+            currentResult = ValueHelper.Less(left, exprs[i]);
+            break;
+          case CompareOperator.Great:
+            currentResult = ValueHelper.Great(left, exprs[i]);
+            break;
+          case CompareOperator.LessEqual:
+            currentResult = ValueHelper.LessEqual(left, exprs[i]);
+            break;
+          case CompareOperator.GreatEqual:
+            currentResult = ValueHelper.GreatEqual(left, exprs[i]);
+            break;
+          case CompareOperator.EqualEqual:
+            currentResult = left.Equals(exprs[i]);
+            break;
+          case CompareOperator.NotEqual:
+            currentResult = !left.Equals(exprs[i]);
+            break;
+          default:
+            throw new System.NotImplementedException(
+                $"Unsupported compare operator: {compare.Ops[i]}");
+        }
+        if (!currentResult) {
+          break;
+        }
+      }
+      _expressionResult = new BooleanValue(currentResult);
+      if (!_visualizerCenter.ComparePublisher.IsEmpty()) {
+        // Evaluate short circuit expressions for visualizers notification.
```

Comment:
```suggestion
        // Evaluates short circuit expressions for visualizers notification.
```

Reply:
I have another idea about this event. These short circuit expressions shall not be evaluated, so that users can be indicated that some expressons are short circuited, and not evaluated.

Reply:
Please have a review again for this.